frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx

```diff
@@ -321,3 +305,25 @@ const maybeAssoc: typeof assoc = (collection, key, value) => {
 
   return assoc(collection, key, value);
 };
+
+// The approximate number of maximum characters for values could be around 7
+// because the values will be compact format when the average length of
+// compact format is less than the average length of full format by 3 characters.
+// https://github.com/metabase/metabase/blob/96e9febfb31d231e3ef08ae907fb42773e065ca5/frontend/src/metabase/static-viz/components/XYChart/Values/Values.tsx#L175
+// e.g. 110,000 vs 100K = 4 !< 7 - 3 it will use full format in this case.
+// So, we could estimate that 7 is the maximum number of characters for values.
+// Then 3.5 is the half the length of the approximate maximum value.
+const APPROXIMATE_MAX_VALUE_CHAR_LENGTH = 7;
+const MAX_SERIES_LENGTH = 15;
+function getValuesLeftOffset(
```

Comment:
This takes care of adding left padding when necessary (having a lot of data points so that the first data point is close to the left y-axis)

#### Results
![image](https://user-images.githubusercontent.com/1937582/187415946-7961bb59-c9d5-4c7c-b41c-c7b6858fc329.png)